lib/Twig/Extension/Core.php

```diff
@@ -1277,7 +1277,15 @@ function twig_lower_filter(Twig_Environment $env, $string)
     function twig_title_string_filter(Twig_Environment $env, $string)
     {
         if (null !== ($charset = $env->getCharset())) {
-            return mb_convert_case($string, MB_CASE_TITLE, $charset);
+            $title = mb_convert_case($string, MB_CASE_TITLE, $charset);
+            $title = preg_replace('#\'S#u', '\'s', $title);
+
+            return preg_replace_callback(
+                            '#\b\d+([^\d\s]+)\b#u',
+                            function ($matches) {
```

Comment:
I agree :)
However as was pointed out in https://github.com/twigphp/Twig/issues/1625 this bug is open for 6 years... Maybe someone else can get the developers @ php to pick this one up?